ocaml/idl/datamodel_vtpm.ml

```diff
@@ -0,0 +1,98 @@
+(*
+   Copyright (C) Citrix Systems Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let persistence_backend =
+  Enum ("persistence_backend", [("xapi", "This VTPM is persisted in XAPI's DB")])
+
+let create =
+  call ~name:"create"
+    ~lifecycle:
+      [
+        (Published, rel_rio, "")
```

Comment:
In reality, the VTPM class and anything in it wasn't actually "published" in that it has been completely unusable until now. The class was there in the datamodel, but the create and destroy functions immediately raise NOT_IMPLEMENTED. No one will have a xapi DB with VTPM rows in it and state.db files will always have ` <table name="VTPM"/>`.

I would therefore consider this class to be introduced _now_, and as a prototype. This signals that any part of the class can still change.

I'd therefore like to clear all `lifecycles` in this module completely. Recording changes from Rio here doesn't really add much value.

Reply:
Cool, I'll change that.